app/uk/gov/hmrc/economiccrimelevyregistration/controllers/deregister/DeregistrationController.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.controllers.deregister
+
+import play.api.libs.json.JsValue
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.actions.AuthorisedAction
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.{BaseController, ErrorHandler}
+import uk.gov.hmrc.economiccrimelevyregistration.models.deregister.Deregistration
+import uk.gov.hmrc.economiccrimelevyregistration.services.deregister.DeregistrationService
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton()
+class DeregistrationController @Inject() (
+  cc: ControllerComponents,
+  deregistrationService: DeregistrationService,
+  authorise: AuthorisedAction
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc)
+    with BaseController
+    with ErrorHandler {
+
+  def upsertDeregistration: Action[JsValue] = authorise(parse.json).async { implicit request =>
+    withJsonBody[Deregistration] { deregistration =>
+      (for {
+        updatedRegistration <- deregistrationService.upsertDeregistration(deregistration).asResponseError
+      } yield updatedRegistration).convertToResult(OK)
```

Comment:
We don't need to return the object sent in the request, can we just `yield ()` (return unit) and NO_CONTENT

Reply:
Fixed.

Reply:
See other comment about NO_CONTENT @paddynski-opencast

Reply:
Changed `upsert` and `delete` to return `NO_CONTENT`.